src/core/hooks.ts

```diff
@@ -1,85 +1,78 @@
+import { fileURLToPath } from 'url'
 import { resolve, posix } from 'path'
-import merge from 'lodash.mergewith'
-import { DefinePlugin } from 'webpack'
+import { defu } from 'defu'
+import type { Consola } from 'consola'
+// import { DefinePlugin } from 'webpack'
+import type { Nuxt } from '@nuxt/schema'
+import { addPluginTemplate, addTemplate, addWebpackPlugin } from '@nuxt/kit'
+import type { Configuration as WebpackConfig } from 'webpack'
+import type { SentryCliPluginOptions } from '@sentry/webpack-plugin'
+import type { Options } from '@sentry/types'
 import * as Sentry from '@sentry/node'
-import { canInitialize, clientSentryEnabled, envToBool, serverSentryEnabled } from './utils'
-import { resolveRelease, resolveClientOptions, resolveServerOptions } from './options'
-
-const RESOLVED_RELEASE_FILENAME = 'sentry.release.config.js'
-
-/**
- * Resolves the options and creates the plugins and the templates at build time.
- *
- * @param      {ThisParameterType<import('@nuxt/types').Module>} moduleContainer
- * @param      {import('../../types/sentry').ResolvedModuleConfiguration} moduleOptions The module options
- * @param      {import('consola').Consola} logger The logger
- * @return     {Promise<void>}
- */
-export async function buildHook (moduleContainer, moduleOptions, logger) {
+import type { ModuleConfiguration, SentryHandlerProxy } from '../../types'
+import { clientSentryEnabled, serverSentryEnabled, envToBool, canInitialize } from './utils'
+import { resolveRelease, ResolvedClientOptions, resolveClientOptions, ResolvedServerOptions, resolveServerOptions } from './options'
+
+const RESOLVED_RELEASE_FILENAME = 'sentry.release.config.mjs'
+
+export async function buildHook (nuxt: Nuxt, moduleOptions: ModuleConfiguration, logger: Consola): Promise<void> {
   const release = await resolveRelease(moduleOptions)
 
+  const templateDir = fileURLToPath(new URL('./templates', import.meta.url))
```

Comment:
When `dist` is mocked then this doesn't work because `templates` directory is one directory up from where `hooks.ts` is in. But when `dist` is built then this code is merged into `module.mjs` so then it works.

Reply:
move files from `core` to the root.